gitops/snapshot.go

```diff
@@ -681,23 +683,24 @@ func PrepareSnapshot(adapterClient client.Client, ctx context.Context, applicati
 		if containerImage == "" {
 			log.Info("component cannot be added to snapshot for application due to missing containerImage", "component.Name", applicationComponent.Name)
 			continue
+		} else {
+			// if the containerImage doesn't have a valid digest, the component
+			// will not be added to snapshot
+			err := ValidateImageDigest(containerImage)
+			if err != nil {
+				log.Error(err, "component cannot added to snapshot for application due to invalid digest in containerImage", "component.Name", applicationComponent.Name)
+				return nil, errors.Join(helpers.NewInvalidImageDigestError(component.Name, containerImage), err)
```

Comment:
I realize we need to call `h.IsInvalidImageDigestError(err)` in https://github.com/redhat-appstudio/integration-service/pull/620/files#diff-887649ea3498038f46836e7fb616f809a6e09232f7fdec4b833f8d7aa8044db3R144, so we can't make errors joined and so I update `NewInvalidImageDigestError` to 
```
func NewInvalidImageDigestError(componentName, image_digest, errMsg string) error {
	return &IntegrationError{
		Reason:  ReasonInvalidImageDigestError,
		Message: fmt.Sprintf("%s is invalid container image digest from component %s: %s", image_digest, componentName, errMsg),
	}
}
```
Please help to review it again, thanks!